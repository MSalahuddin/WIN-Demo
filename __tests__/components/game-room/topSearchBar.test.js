import React from 'react';
import { render } from 'react-native-testing-library';

import TopSearchBar from '../../../js/components/game-room/TopBarSearch';
import MockProvider from '../../MockProvider';

describe('Component BottomNavigator', () => {
  const onSearch = jest.fn();
  const onSelect = jest.fn();
  const onExploreTap = jest.fn();
  test('renders regular top bar correctly', () => {
    const output = render(
      <MockProvider>
        <TopSearchBar
          onSearch={onSearch}
          onSelect={onSelect}
          categoryIndex={1}
          onExploreTap={onExploreTap}
        />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });



});
