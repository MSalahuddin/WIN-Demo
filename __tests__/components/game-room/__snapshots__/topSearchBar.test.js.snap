// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component BottomNavigator renders regular top bar correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#311CA8",
        "flexDirection": "column",
        "marginVertical": 20,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "marginVertical": 16,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 12,
            "left": 12,
            "right": 12,
            "top": 12,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#751EEE",
            "borderRadius": 37.5,
            "marginHorizontal": 24,
            "marginTop": 0,
            "opacity": 1,
            "paddingHorizontal": 15,
            "paddingVertical": 15,
          }
        }
        testID="search-button"
      >
        <Image
          opacity={1}
          resizeMode="contain"
          size={16}
          source={1}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexDirection": "row",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
            "paddingVertical": 10,
          },
        ]
      }
    >
      <RCTScrollView
        data={null}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={1}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        showsHorizontalScrollIndicator={false}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
